robot/impl/resource_manager.go

```diff
@@ -814,15 +833,20 @@ func (manager *resourceManager) processResource(
 			"name", resName, "old_model", gNode.ResourceModel(), "new_model", conf.Model)
 	}
 
-	manager.logger.CDebugw(ctx, "rebuilding", "name", resName)
-	if err := r.manager.closeResource(ctx, currentRes); err != nil {
+	manager.logger.CDebugw(
+		ctx,
+		"rebuilding resource, closing and removing existing resource from graph node",
+		"name", resName,
+		"old_model", gNode.ResourceModel(),
+		"new_model", conf.Model,
+	)
+	if err := r.manager.closeAndUnsetResource(ctx, gNode); err != nil {
 		manager.logger.CError(ctx, err)
 	}
 	newRes, err := r.newResource(ctx, gNode, conf)
 	if err != nil {
-		gNode.UnsetResource()
```

Comment:
I assume there's no consequence of moving the `UnsetResource` call to be before `newResource`?

Reply:
there shouldn't, newResource doesn't care if the node has a resource or not